### PR TITLE
make auc op compatible with 1 dim

### PR DESCRIPTION
**Pull request by guru4elephant**
previous auc calculation method
```python
y_predict = fluid.layers.fc(input=pooled, size=1, act='sigmoid')
loss = fluid.layers.log_loss(input=y_predict, label=fluid.layers.cast(label,dtype='float32'))
avg_cost = fluid.layers.mean(x=loss)
inputs = fluid.layers.concat(input=[1-y_predict, y_predict], axis=1)
auc_out, auc_batch_out, auc_status = fluid.layers.auc(input=inputs, label=label)
```

after this pr
```python
y_predict = fluid.layers.fc(input=pooled, size=1, act='sigmoid')
loss = fluid.layers.log_loss(input=y_predict, label=fluid.layers.cast(label,dtype='float32'))
avg_cost = fluid.layers.mean(x=loss)
auc_out, auc_batch_out, auc_status = fluid.layers.auc(input=y_predict, label=label)
```

test=develop

### CODE REVIEW

**Comment by seiriosPlus** [comment]
Add a UT?

**Comment by seiriosPlus** [comment]
https://github.com/PaddlePaddle/Paddle/blob/de22215c8f8cc272658f56f31ed9ac8efe46d514/paddle/fluid/operators/metrics/auc_op.cc#L31
This line needs to be modified also.

**Review by seiriosPlus — APPROVED** [review]
LGTM
